src/daria/corrections/shape/curvature.py

```diff
@@ -473,3 +349,227 @@ def compute_stretch(self, **kwargs):
             vertical_stretch,
             vertical_stretch_center_offset,
         )
+
+    # ! ---- Main correction routines
+
+    def __call__(
+        self, img: Union[str, np.ndarray], update_cache: bool = False
+    ) -> np.ndarray:
+        """
+        Call method of the curvature correction.
+
+        Applies the curvature correction to a provided image, and returns the
+        corrected image as an array.
+
+        Arguments:
+            img (np.ndarray): image array
+            update_cache (bool): flag controlling whether the transformed coordinates are
+                precomputed even if already existent; default is False.
+
+        Returns:
+            np.ndarray: curvature corrected image.
+        """
+        # FIXME: I suggest to remove the possibility of calling images via str outside
+        # of daria.Image. Opening an image comes with a pre-defined choice of the
+        # colorspace. This control is lost if we do not further pass this information.
+        # Also, modular thinking has had its success in software development for a reason.
+        # I would also say, either the user is loading the image him/herself but then
+        # is required to provide all other necessary information. Other than that, I am
+        # in favor of centralizing opening images from file. This be either in daria.Image
+        # or if need in utils (in addition). But I think, only daria.Image is the right
+        # place for it.
```

Comment:
I agree, but then we need to find a good and easy way of "making" the correction. Oncle we have the correction config file, everything else is taken care of by daria.Image anyways.

Reply:
OK, then I suggest, I add a comment in the documentation here that the option of providing a path should be used with care etc.

Reply:
Ok, that sounds good!